src/helper.jl

```diff
@@ -165,3 +165,181 @@ function writetable(filename::AbstractString; rewrite::Bool=false, kw...)
     writexlsx(filename, xf, rewrite=rewrite)
     nothing
 end
+
+"""
+    emptyfile(sheetname::AbstractString="")
+
+Returns an empty, writable `XLSXFile` with 1 worksheet.
+
+`sheetname` is the name of the worksheet, defaults to `Sheet1`.
+"""
+function emptyfile(sheetname::AbstractString="")
+    xf = open_default_template()
+
+    if sheetname != ""
+        rename!(xf[1], sheetname)
+    end
+
+    return xf
+end
+
+function Base.open(f::Function, filename::AbstractString, sheetname::AbstractString="")
+    xf = open_default_template()
+
+    if sheetname != ""
+        rename!(xf[1], sheetname)
+    end
+
+    try
+        f(xf)
+    finally
+        closefile(xf, filename)
```

Comment:
Probably best not to write a file when an error occurs. Especially since this will overwrite an existing file